### DISCUSSION TITLE
Issues in understanding include directive

### DISCUSSION THREAD

**Discussion by szsoftware** [typst/typst#2083] [General]
Due to the freshness of Typst and lack of more comprehensive examples I'm currently running through some simple issues.

Here: Defining common stuff and include in main document.

This is my code producer, let's say "settings.typ":
`
#set page(
  "a4",
  margin: (
    top: 1cm,
    bottom: 1cm,
    left: 1cm,
    right: 1cm
  )
)`

This is my consumer, let's say "Main.typ":
`
#include "common/settings.typ"

Some Content...
`

I expect having the page set properly, but get an empty page before the content starts, having margins not set as defined.

I'm sure there's just a simple misunderstanding on my side about how the things work together.

Thanks a lot for the great idea (of Typst), path of development and support!

**Comment by freundTech** [comment]
`#include` includes a file with it's own context. This means things like `#set` and `#show` rules don't affect the including document, but only the things in the included document.

You want to write and use a template:

`template.typ`
```typst
#let my-template(body) = {
  set page( "a4", margin: ( top: 1cm, bottom: 1cm, left: 1cm, right: 1cm ) )
  body
}
```
`main.typ`
```typst
#import "template.typ": my-template

#show: my-template

Some content
```

EDIT: Fixed typos in the code

**Comment by freundTech** [comment]
Also see https://typst.app/docs/tutorial/making-a-template/

**Comment by szsoftware** [comment]
Thank you so much for the quick reply and pointing to the template docs!

Also, the contained context/scope hint was/is very important to me.
